### PR TITLE
Fix validation and UI boundary bugs

### PR DESCRIPTION
**Pull request by KillariDev**
## Summary

This PR fixes the ten boundary and validation bugs found during the codebase audit:

1. **EIP-712 fixed-array lengths:** Rejects arrays that are shorter or longer than their declared fixed length instead of accepting undersized values.
2. **EIP-712 required fields:** Rejects primary, domain, and nested struct values when fields declared by their type are missing.
3. **EIP-712 multidimensional arrays:** Descends through dimensions from the correct outermost, rightmost suffix so unequal dimensions are validated in the proper order.
4. **EIP-712 visualization compatibility:** Recognizes one-dimensional fixed-array suffixes during typed-data visualization instead of accepting them at verification and rejecting them later.
5. **EIP-712 domain ordering:** Returns a normal validation failure for incorrectly ordered domain fields instead of throwing an internal exception.
6. **Weighted percentiles:** Replaces shifted cumulative-weight indexing with weighted bucket selection so the 100th percentile cannot index past the dataset.
7. **Ethereum timestamps:** Rejects empty and out-of-range hexadecimal timestamps, invalid `Date` values, and dates before the Unix epoch.
8. **Aave warning metadata:** Corrects the stored key from `app.aave.com/` to the hostname origin `app.aave.com`, allowing its simulation-mode warning to appear.
9. **Tooltip timer races:** Cancels obsolete dismissal timers, prevents an older timer from clearing newer tooltip content, and honors an explicit zero duration.
10. **Dynamic scrolling offsets:** Clamps short-list offsets to zero and guards the unmeasured state so address-book entries are not translated above the viewport or assigned invalid offsets.

Focused regression tests cover every corrected behavior, including malformed and valid EIP-712 cases, percentile boundaries, timestamp codecs, website metadata lookup, tooltip scheduling, and virtual scrolling calculations.

## Review follow-up

The multidimensional EIP-712 review finding is addressed end to end:

- Parses every array suffix in outer-to-inner order, including mixed dynamic and fixed dimensions.
- Recursively validates all dimensions and rejects zero, leading-zero, and malformed bracket suffixes.
- Recursively extracts multidimensional primitive and struct arrays without flattening their shape or treating inner arrays as structs.
- Adds a recursive `nestedArray` enriched-message variant, wire-codec support, and PersonalSign rendering.
- Adds regressions for `string[][]`, valid and invalid multidimensional fixed arrays, malformed dimension syntax, malicious struct-name collisions, extraction shape, and codec round trips.

## Validation

- `bun run test` — passed, including 49 EIP-712 tests
- `bun run setup-chrome` — passed
- `bun run typecheck` — passed
- `bun run lint` — passed

`bun run test:chrome-communication` was not run because these changes do not affect extension messaging or content-script registration.

## Review

The project reviewer’s final pass scored the diff **95/100** and reported no High, Medium, or Low findings. Earlier passes identified an isolated codec-test mismatch and malformed array-dimension edge cases; all were fixed and re-reviewed.

### CODE REVIEW

File: app/ts/utils/eip712Parsing.ts [hunk 1]
```diff
@@ -10,9 +10,12 @@ function findType(name: string, types: readonly { readonly name: string, readonl
 }
 
 function separateArraySuffix(typeWithMaybeArraySuffix: string) {
-	const splitted = typeWithMaybeArraySuffix.split('[]')
-	if (splitted.length === 1) return { arraylessType: typeWithMaybeArraySuffix, isArray: false }
-	return { arraylessType: splitted[0], isArray: true }
+	const arrayMatch = /^(.*)\[(\d*)\]$/.exec(typeWithMaybeArraySuffix)
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply app/ts/utils/eip712Parsing.ts]: Addressed in 17b3330f. Array type parsing now consumes every suffix and recursively validates/extracts each dimension, so `string[][]` and deeper arrays reach the visualization path without being mistaken for structs. The enriched payload preserves nested shape through a recursive `nestedArray` variant with codec and PersonalSign rendering support. Regression coverage includes dynamic multidimensional extraction, per-dimension fixed lengths, malformed/zero/leading-zero suffixes, matching malformed struct keys, and a wire-codec round trip. The full required test/build/typecheck/lint suite passes.

**Review by MicahZoltu — APPROVED** [review]
